### PR TITLE
Configure access for administrator features

### PR DESCRIPTION
**Pull request by vubo**
Configure access for volunteers (return forbidden code for each administrator functionality that a volunteer tries to access).

Pep8 views.py for event, shift, job, organization

https://github.com/systers/vms/issues/87

image

### CODE REVIEW

File: vms/shift/views.py [hunk 1]
```diff
@@ -256,25 +457,44 @@ def list_shifts_sign_up(request, job_id, volunteer_id):
...
+    # check that an admin is logged in
+    if not admin:
+        return HttpResponse('<h1>You don\'t have administrator rights</h1>')
+    else:
+        if volunteer_id:
+            volunteer = get_volunteer_by_id(volunteer_id)
+            if volunteer:
+                # show only shifts that have no hours logged yet
+                # (since it doesn't make sense be able to cancel shifts that have already been logged)
+                shift_list = get_unlogged_shifts_by_volunteer_id(volunteer_id)
```
Comment (tapasweni-pathak) [inline vms/shift/views.py]: You comment very well @vubo. :+1:

**Comment by tapasweni-pathak** [comment]
@vubo How about not simply returning a HttpResponse and putting it as a header. Can we do something better in the view? We can put one html file in templates.

How about having a back button in 403 page and also the response string given in the same style as we are having effects (for text) everywhere else?

**Comment by vubo** [comment]
@tapasweni-pathak thank you for the suggestion, I did it. Now it looks like this:

image

**Comment by tapasweni-pathak** [comment]
@vubo One more thing, instead of making different files for volunteers accessing administrator rights and  the opposite. How about making a common file, naming it, 403_forbidden or something, and printing, Forbidden, you don't have the rights.

Being specific is nice. But I'm thinking of having one file for all errors, 403 error and using it everywhere, What do you think? I'm not sure, what will be the best practice.

@willingc @rosariorobinson  What do you think?

**Comment by willingc** [comment]
@vubo I'm sorry that I've been out of touch the past week and a half. I plan to spend some time tomorrow reviewing the PRs and progress that you have been making with @tapasweni-pathak's guidance. 

@tapasweni-pathak Thanks for being so responsive :D I will look at this PR tomorrow.

**Comment by vubo** [comment]
@willingc all is well, thank you!

@tapasweni-pathak I'm not sure, but I think a couple of specific pages will be better than one general. So it is clear to the user. And it's not so much work.

**Comment by tapasweni-pathak** [comment]
@vubo Yes, as I said being specific is nice. But yes, anyway as we have different outputs on our screen, contributors will get to know that we have different files for showing 403 errors while understanding the code. 

I'm merging this then. :+1: :dancer:

**Comment by tapasweni-pathak** [comment]
@vubo You have not rebased your latest changes?

**Comment by vubo** [comment]
@tapasweni-pathak I did it, thank you!

Do I always have to rebase? We can not merge a PR with 2 commits?

**Comment by tapasweni-pathak** [comment]
Yes, rebasing is always good, if some suggestions are made on first PR, or on some commit. We don't want to commit something which we did not find good in the first place. When someone wants to contribute every commit should be easy to track.

**Comment by vubo** [comment]
@tapasweni-pathak now I understand, thank you!

**Pull request merged**